### ISSUE TITLE
best configuration for reconstructing from >100 images

### ISSUE THREAD

**Issue by zcczhang** [naver/mast3r#21]
Hi authors, thank you for bringing such a great project out!

For reconstructing scenes with hundreds of images (like a video in the wild), what are the best configurations for the reconstruction? Would the symmetrizing and "complete" scene graph give too many pairs? 

Before, in DUSt3R, I could subsample a small ratio from n*(n-1) pairs and do the reconstruction, and the results looked great, but in MASt3R, it will throw the key error [here](https://github.com/naver/mast3r/blob/d1c9c7baf029d7d0890969391c6621ce2ed42a83/mast3r/cloud_opt/sparse_ga.py#L320) due to the subsampled pairs.

**Comment by yocabon** [comment]
Hi,
we are still in the process of improving the sparse_ga for large image collections.
that being said, it should still work with a subsample of pairs: logwin would be recommended (to have short and long range connections between images).
I do not know why you are getting an error, could you tell me more ?

**Comment by zcczhang** [comment]
yeah I've tried with logwin with a few different window sizes and seems the results are not as good as using much fewer images, but that may also be due to the other fitting parameters. 

In terms of the error, here is the pseudocode I used:
```{python}
subsample_pair_ratio = 0.1
pairs = make_pairs(imgs, scene_graph="complete", prefilter=None, symmetrize=True)
n_pairs =  int(len(pairs) * subsample_pair_ratio)
pairs = random.sample(pairs, n_pairs)
``` 
and the do `sparse_global_alignment(
                filelist,
                pairs,
               , ... )`  for alignment. Then the code I pointed before will raise the Keyerror for `is_matching_ok[img1, img2]:`, where I am guessing that's just the unsampled/unused image pairs?

**Comment by yocabon** [comment]
The error most likely come from you breaking the symmetry (having both img1, img2 and img2, img1 in the pairs) when you do random.sample(pairs, n_pairs).

**Comment by zcczhang** [comment]
Yeah I thought so but seems to raise the same error even passing `symmetrize=False`

**Comment by yocabon** [comment]
and did you reapply symmetry after random.sample ? - pairs += [(img2, img1) for img1, img2 in pairs]

**Comment by migsdigs** [comment]
@zcczhang Were you able to get something like this working using a subsample of pairs? I am trying to do something similar
